ARROW-13716: Add RecordBatch recipe (#66)
apache/arrow-cookbook: python/source/create.rst

    :func:`pyarrow.array` for conversion to Arrow arrays,
    and will benefit from zero copy behaviour when possible.

> Creating RecordBatches

Reviewer: Nit: Given the titles are prose I think it might be more correct to separate the words.
```suggestion
Creating Record Batches
```

======================

> Most IO operations in Arrow happen shipping batches of data

Reviewer: ```suggestion
Most I/O operations in Arrow happen when shipping batches of data
```

> to the destination, :class:`pyarrow.RecordBatch` are the way

Reviewer: ```suggestion
to the destination.  :class:`pyarrow.RecordBatch` is the way
```

> to the destination.  :class:`pyarrow.RecordBatch` is the way

Reviewer 2: ```suggestion
to their destination.  :class:`pyarrow.RecordBatch` is the way
```

> Arrow represents batches of data, they can be seen as a slice

Reviewer 2: ```suggestion
Arrow represents batches of data.  A RecordBatch can be seen as a slice
```

of a table.

.. testcode::

    import pyarrow as pa

    batch = pa.RecordBatch.from_arrays([
        pa.array([1, 2, 3, 4, 5]),
        pa.array([10, 20, 30, 40, 50])

>     ], names=["first", "second"])

Reviewer: Minor nit: Since we are introducing the concept of record batches here it may be clearer to the user if the column names and values are meaningful.  As it stands it might not be immediately obvious that `first` corresponds to `[1, 2, 3, 4, 5]` and `second` corresponds to `[10, 20, 30, 40, 50]`.  However, if the example were real world data it may be more clear.

Author: Refactored the example to use odd and even numbers, should be immediately clear to the reader of things map

> multiple batches can be combined into a table using 

Reviewer: ```suggestion
Multiple batches can be combined into a table using 
```

:meth:`pyarrow.Table.from_batches`

.. testcode::

    second_batch = pa.RecordBatch.from_arrays([
        pa.array([6, 7, 8, 9, 10]),
        pa.array([60, 70, 80, 90, 100])
    ], names=["first", "second"])

    table = pa.Table.from_batches([batch, second_batch])

.. testcode::

    print(table)

.. testoutput::

    pyarrow.Table
    first: int64
    second: int64

> Equally, :class:`pyarrow.Table` can be converted to a set of 

Reviewer: ```suggestion
Equally, :class:`pyarrow.Table` can be converted to a list of 
```
